examples/with-nextjs-and-clerk-auth/src/components/ThemeRegistry/RootThemeProvider.tsx

```diff
@@ -1,25 +1,131 @@
 'use client';
 
 import * as React from 'react';
+import { PropsWithChildren, useCallback, useMemo } from 'react';
 
+import { useAuth } from '@clerk/nextjs';
+import { Theme } from '@emotion/react';
 import CssBaseline from '@mui/material/CssBaseline';
+import { createTheme } from '@mui/material/styles';
 import { ThemeProvider } from '@mui/material/styles';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import * as themes from '@team-monite/sdk-themes';
+
+import {
+  PrivateMetadata,
+  SelectedTheme,
+  ThemeMode,
+  ThemeVariant,
+} from '@/lib/clerk-api/types';
 
 import NextAppDirEmotionCacheProvider from './EmotionCache';
-import { theme } from './theme';
 
-/**
- * Provides the theme to the Root Application
- * Does not affect the Monite SDK.
- */
-export function RootThemeProvider({ children }: { children: React.ReactNode }) {
+type Context = {
+  setThemeMode: (mode: ThemeMode) => void;
+  setThemeVariant: (theme: ThemeVariant) => void;
+  selectedTheme: SelectedTheme;
+  theme: Theme;
+};
+
+type RootThemeProviderProps = PropsWithChildren<{
+  initialTheme?: SelectedTheme;
+}>;
+
+const QUERY_KEY = ['theme'];
+const METADATA_ROUTE = '/api/user/theme';
+
+export const RootThemeProviderContext = React.createContext<
+  Context | undefined
+>(undefined);
+
+export function RootThemeProvider(props: RootThemeProviderProps) {
```

Comment:
If you don't mind, let's rename to `AppThemeProvider`, I feel like that would be a more understandable name, though I'm not at all sure.